src/util/APIService.js

```diff
@@ -0,0 +1,88 @@
+import axios from 'axios';
+import { BACKEND_URL } from './consts';
+
+export default class APIService {
+	static url = BACKEND_URL;
+
+	static data = {};
+
+	static Get(endpoint, queries = {}) {
+		if (
+			!(
+				queries &&
+				Object.keys(queries).length === 0 &&
+				queries.constructor === Object
+			)
+		) {
+			return new Promise((resolve, reject) => {
+				const payload = Object.assign(this.data, queries);
+				const esc = encodeURIComponent;
+				const query = Object.keys(payload)
+					.map((k) => `${esc(k)}=${esc(payload[k])}`)
+					.join('&');
+				try {
+					const response = axios
+						.get(`${this.url + endpoint}/?${query}`)
+						.then((res) => resolve(res.data.result))
+						.catch((err) => reject(err));
+					resolve(response);
+				} catch {
+					const response = 'failed to get';
+					reject(response);
+				}
+			});
+		}
+		return new Promise((resolve, reject) => {
+			try {
+				axios
+					.get(`${this.url + endpoint}`)
+					.then((res) => resolve(res.data.result))
+					.catch((err) => reject(err));
+			} catch {
+				const response = 'failed to get';
+				reject(response);
+			}
+		});
+	}
+
+	static Post(endpoint, rawdata = {}, token = '') {
+		return new Promise((resolve, reject) => {
+			try {
+				axios({
+					method: 'post',
+					url: `${this.url + endpoint}`,
+					data: rawdata,
+					headers: { Authorization: token },
+				})
+					.then((res) => {
+						resolve(res.data);
+					})
+					.catch((err) => {
+						reject(err);
+					});
+			} catch {
+				const response = 'failed to post';
+				reject(response);
+			}
+		});
+	}
+
+	static DELETE(endpoint, specifier = '', token = '') {
+		return new Promise((resolve, reject) => {
+			try {
+				axios({
```

Comment:
axios methods already return promise so your own Promise and try catch blocks is redundant.